### PR TITLE
Create the InputParser class

### PR DESCRIPTION
**Pull request by osvegn**
This class will be used to parse the arguments of a command of pizza while the program is running
https://github.com/Cavonstavant/PLAZAAAAAAAAAAAAAAAAAAAAAAA/issues/4

### CODE REVIEW

**Comment by osvegn** [comment]
Wait while this https://github.com/Cavonstavant/PLAZAAAAAAAAAAAAAAAAAAAAAAA/pull/14 is open

File: src/Parser/InputParser/InputParser.cpp [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    ParserEX("Get the command for the InputParser class", Logger::INFO);
+    return (_command);
+}
+
+void InputParser::prossessArguments()
+{
+    std::vector<std::string> tmp = _arguments;
+    std::vector<std::string> args;
+
+    ParserEX("Prossess the arguments for the InputParser class", Logger::INFO);
```
Comment (Cavonstavant) [inline src/Parser/InputParser/InputParser.cpp:52]: same

File: src/Parser/InputParser/InputParser.cpp [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+            "Americana",
+            "Fantasia"};
+    bool findValidePizza = false;
+
+    for (std::size_t index = 0; index < 4; index++) {
+        if (availablePizza[index] == pizza)
+            findValidePizza = true;
+    }
+    if (!findValidePizza)
+        ParserEX("Invalide pizza name in the command", Logger::LOW);
```
Comment (Cavonstavant) [inline src/Parser/InputParser/InputParser.cpp:80]: Already better, Invalid w/o an e would be ideal
Consider adding the faulty pizza name to `what()`

File: src/Parser/InputParser/InputParser.cpp [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+        for (auto arg: _arguments) {
+            args.push_back(arg);
+        }
+        args.push_back(";");
+    }
+    _arguments = args;
+    _checkArgumentsValidity();
+}
+
+bool InputParser::_isValidePizzaName(const std::string &pizza) const
```
Comment (Cavonstavant) [inline src/Parser/InputParser/InputParser.cpp:66]: Valid not valide

File: src/Parser/InputParser/InputParser.cpp [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+    for (std::size_t index = 0; index < 4; index++) {
+        if (availablePizza[index] == pizza)
+            findValidePizza = true;
+    }
+    if (!findValidePizza)
+        ParserEX("Invalide pizza name in the command", Logger::LOW);
+    return (findValidePizza);
+}
+
+bool InputParser::_isValidePizzaSize(const std::string &pizzaSize) const
```
Comment (Cavonstavant) [inline src/Parser/InputParser/InputParser.cpp:84]: valid not valide

File: src/Parser/InputParser/InputParser.cpp [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+            "XL",
+            "XXL"};
+    bool findValidePizzaSize = false;
+
+    for (std::size_t index = 0; index < 5; index++) {
+        if (availablePizzaSize[index] == pizzaSize)
+            findValidePizzaSize = true;
+    }
+    if (!findValidePizzaSize)
+        ParserEX("Invalide pizza size in the command", Logger::LOW);
```
Comment (Cavonstavant) [inline src/Parser/InputParser/InputParser.cpp:99]: Same as the pizza name

File: src/Parser/InputParser/InputParser.cpp [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+    bool findValidePizzaNumber = false;
+    std::string tmp = pizzaNumber;
+
+    if (tmp.find("x") == 0) {
+        tmp.substr(1);
+        if (std::atoi(tmp.c_str()) != 0)
+            findValidePizzaNumber = true;
+    }
+    if (!findValidePizzaNumber)
+        ParserEX("Invalide pizza number in the command", Logger::LOW);
```
Comment (Cavonstavant) [inline src/Parser/InputParser/InputParser.cpp:114]: same as the name and the size

File: src/Parser/InputParser/InputParser.hpp [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+        ~InputParser() = default;
+        /// \brief Set the command that will be parsed.
+        /// \param const std::string & is the command string
+        void setCommand(const std::string &);
+        /// \brief Get the command previously set.
+        /// \return const std::string& of the command.
+        /// It return an empty std::string in case of undefined command.
+        const std::string &getCommand() const;
+        /// \brief Prossess the argument to set them in a form usefull for the program
+        void prossessArguments() override;
```
Comment (Cavonstavant) [inline src/Parser/InputParser/InputParser.hpp:30]: process not prossess

File: src/Parser/InputParser/InputParser.hpp [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+*/
+
+/// \file src/Parser/InputParser/InputParser.hpp
+
+#ifndef INPUTPARSER_HPP_
+#define INPUTPARSER_HPP_
+
+#include "AParser.hpp"
+
+/// \brief InputParser is a class use to parse command of pizza while the program is running
```
Comment (Cavonstavant) [inline src/Parser/InputParser/InputParser.hpp:15]: used to*

**Review by Cavonstavant — APPROVED** [review]

**Review by Cavonstavant — CHANGES_REQUESTED** [review]
need a bit of rework especially on exception throwing

**Review by Cavonstavant — APPROVED** [review]
LGTM!

File: src/Parser/InputParser/InputParser.hpp [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+        void _checkArgumentsValidity();
+        /// \brief Check if the pizza name passed as parameter is valide.
+        /// It will print a message if an error occured.
+        /// \param const std::string & is the pizza name.
+        /// \return true if it's std::string is a valide pizza name, false otherwise.
+        bool _isValidPizzaName(const std::string &) const;
+        /// \brief Check if the pizza size passed as parameter is valide.
+        /// It will print a message if an error occured.
+        /// \param const std::string & is the pizza size.
+        /// \return true if it's std::string is a valide pizza size, false otherwise.
```
Comment (31Nathan) [inline src/Parser/InputParser/InputParser.hpp:51]: It's "valid"

File: src/Parser/InputParser/InputParser.hpp [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+        bool _isValidPizzaSize(const std::string &) const;
+        /// \brief Check if the pizza number passed as parameter is valide.
```
Comment (31Nathan) [inline src/Parser/InputParser/InputParser.hpp:53]: Same

File: src/Parser/InputParser/InputParser.hpp [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+        /// It will print a message if an error occured.
+        /// \param const std::string & is the pizza number.
+        /// \return true if it's std::string is a valide pizza number, false otherwise.
```
Comment (31Nathan) [inline src/Parser/InputParser/InputParser.hpp:56]: And we continue

File: src/Parser/InputParser/InputParser.hpp [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+        bool _isValidPizzaNumber(const std::string &) const;
+        /// \brief Check if the pizza separator passed as parameter is valide.
```
Comment (31Nathan) [inline src/Parser/InputParser/InputParser.hpp:58]: Not going to do all

File: src/Parser/InputParser/InputParser.hpp [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+        /// It return an empty std::string in case of undefined command.
+        const std::string &getCommand() const;
+        /// \brief Process the argument to set them in a form usefull for the program
+        void processArguments() override;
+        /// \brief Set the separator for the command
+        /// \param const char Separator char
+        void setCommandSeparator(const char);
+
+    private:
+        /// \brief Store the command that will be prossess.
```
Comment (31Nathan) [inline src/Parser/InputParser/InputParser.hpp:36]: It's "process"

File: src/Parser/InputParser/InputParser.hpp [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+        std::string _command;
+        /// \brief Store the separator that will use for prossessArgument.
```
Comment (31Nathan) [inline src/Parser/InputParser/InputParser.hpp:38]: Here we go again

**Review by 31Nathan — APPROVED** [review]

**Pull request merged**